garden/ros_gz_project_template_guide.md

```diff
@@ -0,0 +1,97 @@
+# Guide to `ros_gz_project_template` for ROS 2 and Gazebo Development
+
+In this guide, you will learn how to use the `ros_gz_project_template` to create a (recommended) structured workspace or improve your existing workspace for your ROS 2 and Gazebo projects. This template offers a consistent layout, automated build process, and integration with both ROS 2 and Gazebo, enabling you to focus on developing your robotics applications.
+
+## Installation Steps
+
+1. If you are starting a project, create new workspace or go to your existing project's source directory:
+
+```bash
+    mkdir -p ~/project_ws/src
+    cd ~/project_ws/src
+```
+
+1. Directly `Use this template` and create your project repository on Github.
+
+   ![use_template](tutorials/ros2_integration/use_template.png)
+
+   Or start by cloning the `ros_gz_project_template` repository:
+
+   ```bash
+    wget https://raw.githubusercontent.com/gazebosim/ros_gz_project_template/main/template_workspace.yaml
+    vcs import < template_workspace.yaml
+   ```
+
+1. Rename the cloned repository folder to your desired project name:
+
+   ```bash
+   mv ros_gz_project_template your_project_name
+   ```
+
+   Change your working directory to the project workspace:
+
+   ```bash
+   cd ~/project_ws
+   ```
+
+## Package structure
+
+At this point you'll have the following packages in your project workspace:
+
+* `ros_gz_example_description` - holds the sdf description of the simulated system and any other simulation assets. 
+Simulation assets means your models or robot descriptions in URDF or SDF, meshes and materials files to help visualize different parts of the robot and finally compiling all these elements in a simulated world SDF. Existing assets can be used by installing the models directory and exporting the paths to your environment. 
+Setting up paths can be also automated using ament environment hooks with a DSV file prepending the model share path to Gazebo resource path. 
```

Comment:
This is going to sound very advanced to a beginner. Is there an example we can link to? I link to the colcon page, but that is pretty cryptic. In fact, the dsv files themselves look cryptic without a concrete example.
```suggestion
Setting up paths can be also automated using colcon environment hooks with a [DSV file](https://colcon.readthedocs.io/en/released/developer/environment.html?highlight=dsv#dsv-files) prepending the model share path to Gazebo resource path. 
```